### PR TITLE
fix: issues - 233、232、245、234

### PR DESCRIPTION
**Pull request by devchenyan**
fix: 
- https://github.com/Magickbase/neuron-public-issues/issues/233
- https://github.com/Magickbase/neuron-public-issues/issues/232
- https://github.com/Magickbase/neuron-public-issues/issues/245
- https://github.com/Magickbase/neuron-public-issues/issues/234

### CODE REVIEW

File: packages/neuron-ui/src/components/HistoryDetailPage/index.tsx [hunk 1]
```diff
@@ -49,27 +50,21 @@ const HistoryDetailPage = () => {
   const [t] = useTranslation()
   const [transaction, setTransaction] = useState(transactionState)
   const [error, setError] = useState({ code: '', message: '' })
+  const [faidMessage, setFaidMessage] = useState('')
```
Comment (WhiteMinds) [inline packages/neuron-ui/src/components/HistoryDetailPage/index.tsx:53]: faid -> failed

**Review by WhiteMinds — APPROVED** [review]

File: packages/neuron-ui/src/components/HistoryDetailPage/index.tsx [hunk 2]
```diff
@@ -79,7 +74,7 @@ const HistoryDetailPage = () => {
           })
         })
     }
-  }, [t, hash, currentWallet])
+  }, [t, hash, currentWallet, navigate])
```
Comment (yanguoyu) [inline packages/neuron-ui/src/components/HistoryDetailPage/index.tsx:77]: Why should we add `navigate` here?

File: packages/neuron-ui/src/utils/formatters.ts [hunk 3]
```diff
@@ -329,7 +329,10 @@ export const padFractionDigitsIfDecimal = (num: string | number, minimumFraction
  * -100.00002362 CKB ---> -100.00002362
  * +10,000 CKB ---> 10000
  *  */
-export const complexNumberToPureNumber = (value: string) => {
+export const complexStringToPureNumberOrString = (value: string) => {
```
Comment (Keith-CY) [inline packages/neuron-ui/src/utils/formatters.ts:332]: This method seems dangerous, it parses `100x.a1 CKB` to `100.1`

File: packages/neuron-ui/src/tests/formatters/complexNumberToPureNumber/index.test.ts [hunk 4]
```diff
@@ -1,10 +1,10 @@
-import { complexNumberToPureNumber } from 'utils/formatters'
+import { complexStringToPureNumberOrString } from 'utils/formatters'
 import fixtures from './fixtures'
 
 const fixtureTable: [string, number][] = fixtures.map(({ value, expected }) => [value, expected])
 
 describe(`Verify currency formatter`, () => {
   test.each(fixtureTable)(`%j => %s`, (value: string, expected: number) => {
-    expect(complexNumberToPureNumber(value)).toBe(expected)
+    expect(complexStringToPureNumberOrString(value)).toBe(expected)
```
Comment (Keith-CY) [inline packages/neuron-ui/src/tests/formatters/complexNumberToPureNumber/index.test.ts:8]: The file name should be updated according to the method name change
And more cases(`return value directly`) should be added for the update inside the method

**Comment by silySuper** [comment]
@devchenyan help to push this review.

**Review by yanguoyu — APPROVED** [review]

**Pull request merged**